src/app/view/Secure/PinPrompt.tsx

```diff
@@ -1,18 +1,33 @@
+import { RouteProp } from '@react-navigation/native'
 import React, { useEffect } from 'react'
 
 import { PromptingPage } from '/components/templates/PromptingPage'
 import { navigate } from '/libs/RootNavigation'
 import { routes } from '/constants/routes'
 import { translation } from '/locales'
 import { startPinCode } from '/app/domain/authorization/services/SecurityService'
+import { devlog } from '/core/tools/env'
 
-export const PinPrompt = (): JSX.Element => {
+type RootStackParamList = Record<string, undefined | { onSuccess: () => void }>
+
+interface PinPromptProps {
+  route: RouteProp<RootStackParamList, 'pinPrompt'>
+}
+
+export const PinPrompt = (props: PinPromptProps): JSX.Element => {
   const handleSetPinCode = (): void => {
-    navigate(routes.setPin)
+    navigate(routes.setPin, { onSuccess: props.route.params?.onSuccess })
   }
 
   const handleIgnorePinCode = (): void => {
-    navigate(routes.home)
+    try {
+      if (!props.route.params?.onSuccess)
+        throw new Error('No onSuccess callback given to PinPrompt')
```

Comment:
the `if` condition is different from the one on `PasswordPrompt`, is that wanted?

Reply:
I should consolidate the two functions to be the same (import it)

Reply:
updated